pandas/core/ops.py

```diff
@@ -1264,6 +1272,9 @@ def f(self, other, axis=default_axis, level=None, fill_value=None):
             if fill_value is not None:
                 self = self.fillna(fill_value)
 
+            if is_number(other):
```

Comment:
I think this is better handled in combine_const

Reply:
what do you think would be the best approach in that method? cast bool columns to float/int before eval is used?

Reply:
yes